connectors/services/job_scheduling.py

```diff
@@ -145,34 +147,68 @@ async def _run(self):
         return 0
 
     async def _on_demand_sync(self, connector):
-        if not connector.sync_now:
-            return
-
-        await self.sync_job_index.create(
-            connector=connector, trigger_method=JobTriggerMethod.ON_DEMAND
-        )
-        await connector.reset_sync_now_flag()
+        @with_concurrency_control()
+        async def _should_schedule_on_demand_sync():
+            try:
+                await connector.reload()
+            except DocumentNotFoundError:
+                logger.error(f"Couldn't reload connector {connector.id}")
+                return False
+
+            if not connector.sync_now:
+                return False
+
+            await connector.reset_sync_now_flag()
+            return True
```

Comment:
Maybe we can move this function outside of the function so it can be tested?

Reply:
It was nested because it needs to be wrapped by `@with_concurrency_control`. It's tested together with the outer function `_on_demand_sync`